### PR TITLE
Flex Deployment Changes

### PR DESCRIPTION
**Pull request by khkh-ms**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/Azure.Functions.Cli/Helpers/AzureHelper.cs [hunk 1]
```diff
@@ -423,9 +423,41 @@ public static async Task<Site> LoadSiteObjectAsync(Site site, string accessToken
...
+                return false;
+            }
+
+            runtimeConfig.name = runtimeName;
+            runtimeConfig.version = runtimeValue;
+
+            string functionAppString = JsonConvert.SerializeObject(functionApp);
+
+            url = new Uri($"{managementURL}{site.SiteId}?api-version={ArmUriTemplates.FlexApiVersion}");
+            await ArmHttpAsyncForFlex(new HttpMethod("PATCH"), url, accessToken, functionApp);
```
Comment (kamperiadis) [inline src/Azure.Functions.Cli/Helpers/AzureHelper.cs:448]: AFAIK, we currently can't do a PATCH request to update attributes, we'll need to do a PUT.

Comment (khkh-ms) [reply src/Azure.Functions.Cli/Helpers/AzureHelper.cs:448]: Thanks @kamperiadis!  Can you please share the recommended API. I got it from my local AZ CLI copy which was using Patch. Interestingly, it is actually working in beta :).

Comment (kamperiadis) [reply src/Azure.Functions.Cli/Helpers/AzureHelper.cs:448]: I just confirmed with David today - we did not do any implementation changes for PATCH, which might be missing things like blocking unsupported configs in the Site object. I would not suggest using PATCH. Instead, we should be doing a GET call to the function app and then do a PUT request with that same function app object but with the updated values. 

Not sure what Az CLI copy you have but for flex specifically, the Az CLI makes PUT calls when updating the runtime.

**Review by kamperiadis — CHANGES_REQUESTED** [review]

File: src/Azure.Functions.Cli/Helpers/AzureHelper.cs (hunk 1 above)
Comment (khkh-ms) [reply src/Azure.Functions.Cli/Helpers/AzureHelper.cs]: Changed it to us `PUT` instead of `PATCH`.

**Review by kamperiadis — APPROVED** [review]

**Pull request merged**